src/transformers/models/textnet/modeling_textnet.py

```diff
@@ -0,0 +1,447 @@
+
+        stages = []
+        num_stages = len(config.conv_layer_kernel_sizes)
+        for stage_ix in range(0, num_stages):
+            stages.append(TextNetStage(config, stage_ix))
+
+        self.stages = nn.ModuleList(stages)
+
+    def forward(self, hidden_state):
+        hidden_states = []
+        for stage in self.stages:
+            hidden_state = stage(hidden_state)
+            hidden_states.append(hidden_state)
+
+        return hidden_states
+
+
+class TextNetPreTrainedModel(PreTrainedModel):
+    """
+    An abstract class to handle weights initialization and a simple interface for downloading and loading pretrained
+    models.
+    """
+
+    config_class = TextNetConfig
+    base_model_prefix = "textnet"
+    main_input_name = "pixel_values"
+
+    def _init_weights(self, module):
+        if isinstance(module, (nn.Linear, nn.Conv2d)):
+            module.weight.data.normal_(mean=0.0, std=self.config.initializer_range)
+            if module.bias is not None:
+                module.bias.data.zero_()
+
+
+@add_start_docstrings(
+    "The bare Textnet model outputting raw features without any specific head on top.",
+    TEXTNET_START_DOCSTRING,
+)
+class TextNetModel(TextNetPreTrainedModel):
+    def __init__(self, config):
+        super().__init__(config)
+        self.stem = TextNetConvLayer(config)
+
+        self.encoder = TextNetEncoder(config)
+
+        self.pooler = nn.AdaptiveAvgPool2d((2, 2))
+
+        self.init_weights()
+
+    @add_start_docstrings_to_model_forward(TEXTNET_INPUTS_DOCSTRING)
+    @add_code_sample_docstrings(
+        checkpoint=_CHECKPOINT_FOR_DOC,
+        output_type=BaseModelOutputWithPoolingAndNoAttention,
+        config_class=_CONFIG_FOR_DOC,
+        modality="vision",
+        expected_output=_EXPECTED_OUTPUT_SHAPE,
+    )
+    def forward(
+        self, pixel_values: Tensor, output_hidden_states: Optional[bool] = None, return_dict: Optional[bool] = None
+    ) -> Union[Tuple[Any, List[Any]], Tuple[Any], BaseModelOutputWithPoolingAndNoAttention]:
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+        output_hidden_states = (
+            output_hidden_states if output_hidden_states is not None else self.config.output_hidden_states
+        )
+
+        hidden_state = self.stem(pixel_values)
+        hidden_states = [hidden_state]
+
+        hidden_states = hidden_states + self.encoder(hidden_state)
+
+        last_hidden_state = hidden_states[-1]
+        pooled_output = self.pooler(last_hidden_state)
+
+        if not return_dict:
+            output = (last_hidden_state, pooled_output)
+            return output + (tuple(hidden_states),) if output_hidden_states else output
+
+        return BaseModelOutputWithPoolingAndNoAttention(
+            pooler_output=pooled_output,
+            last_hidden_state=last_hidden_state,
+            hidden_states=tuple(hidden_states) if output_hidden_states else None,
+        )
```

Comment:
this can't be right, cause for the tuple output you first set the last_hidden_state, whereas the dict output has last_hidden_state at the second position

please don't skip the equivalence test

Reply:
Okay, the equivalence test has not been skipped, earlier with the right order, the test was failing (When I swapped order, it passed ), I am fixing the order now, the tests pass.